test/metabase/query_processor_test/parameters_test.clj

```diff
@@ -73,6 +74,18 @@
               (is (= 1
                      (count-with-params :users :last_login :date/single "2014-08-02T09:30Z" options))))))))))
 
+(deftest template-tag-generation-test
+  (testing "Generating template tags produces correct types for running process-query (#31252)"
+    (mt/with-temp* [Card [{card-id :id} {:dataset       true
```

Comment:
Can you use `t2.with-temp/with-temp` here? We're working on phasing out Toucan 1 stuff. The only difference is instead of `[Card [binding properties]]` it's `[Card binding properties]`.